### PR TITLE
Remove simple alias for `wait` -> `settled`.

### LINKED ISSUE #283: Unsafe options usage in isSettled
This is definitely a fault on my end, but something I think is worth bringing up, at least for anyone else who may encounter this.

We have been using the library for years, however a few of our tests were using it incorrectly. This wasn't a problem in the past as it was a bit safer.

Typical usage
```
import wait from 'ember-test-helpers/wait';
...
return wait().then(()=>{
    assert....
})
```

Which is correct usage, however a few of my tests used it incorrectly, for example `wait(3000)` thinking it would also wait 3 seconds. This wasn't a problem in the past as the code in question used `hasOwnProperty` which was safe on a variable which was a number. Now it uses the `in` keyword which breaks on non objects.

We would see errors such as `Uncaught TypeError: Cannot use 'in' operator to search for 'waitForTimers' in 500`

[how it was](https://github.com/emberjs/ember-test-helpers/blob/v0-6-x/lib/ember-test-helpers/wait.js#L55-L57)
[how it is now](https://github.com/emberjs/ember-test-helpers/blob/master/addon-test-support/%40ember/test-helpers/settled.js#L112-L116)

Easiest fix for users: remove the number, and better understand how/what wait is actually doing
Easiest fix for library: explicity check options is an object, and not just undefined

### LINKED ISSUE #300: Arbitrary arguments to `wait` cause error
Seems since this [commit](https://github.com/emberjs/ember-test-helpers/commit/0a83ee3b822f59813baf71bc50938cc71e70f1b4) where `wait` delegates to `settled`, passing a non-object (e.g. boolean) to `wait` would cause a `Cannot use 'in' operator to search for 'waitForTimers' in true` error. Prior to this, this would work (the argument would simply have no effect).

Offending line: https://github.com/emberjs/ember-test-helpers/blob/9eb3e32f8340d01704e950f317b282f07073f1e9/addon-test-support/%40ember/test-helpers/settled.js#L171
Example: https://travis-ci.org/kybishop/ember-popper/jobs/328856339#L547
Caused by: https://github.com/kybishop/ember-popper/blob/5d4457588fd9f30f7a0a2aa34a28e86209ae5978/tests/integration/components/ember-popper/action-test.js#L46

### PR DESCRIPTION
**Pull request by rwjblue**
The original goal of `settled` was to "rebrand" `wait` as `settled` without the 'baggage" (since `wait` had some legacy APIs that should not exist in `settled`). Unfortunately, the code refactoring that I had originally done that allowed them to be shared resulted in introducing bugs like those reported in #283.

This commit completely severs the implementation of `wait` from that of `settled`, by leveraging the (IMHO very nice) layers of abstraction that we have created since the original extraction.  Implementing `wait` with `waitUntil` and `getSettledState` makes this quite trivial now.

tldr of changes:

* `isSettled` no longer has to support the legacy `wait` contract (this was undocumented private behavior _only_ kept around to support the old semantics of `wait`)
* `wait` is implemented as its own `waitUntil` with the same semantics it has had since the 0.6.x series

Fixes #300
Fixes #283
Closes #303

### CODE REVIEW

File: addon-test-support/ember-test-helpers/wait.js [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
+
+/**
+  Returns a promise that resolves when in a settled state (see `isSettled` for
+  a definition of "settled state").
+
+  @private
+  @deprecated
+  @returns {Promise<void>} resolves when settled
+*/
+export default function settled() {
```
Comment (Turbo87) [inline addon-test-support/ember-test-helpers/wait.js]: I think this should be called `wait()`, not `settled()`

File: addon-test-support/ember-test-helpers/wait.js [hunk 2]
```diff
@@ -1,7 +1,46 @@
...
+  let options = arguments[0];
```
Comment (Turbo87) [inline addon-test-support/ember-test-helpers/wait.js]: we should use `function wait(options = {}) {` instead

File: addon-test-support/ember-test-helpers/wait.js (hunk 1 above)
Comment (rwjblue) [reply addon-test-support/ember-test-helpers/wait.js]: DOH! Indeed!

File: addon-test-support/ember-test-helpers/wait.js (hunk 2 above)
Comment (rwjblue) [reply addon-test-support/ember-test-helpers/wait.js]: kk, seems good

Comment (rwjblue) [reply addon-test-support/ember-test-helpers/wait.js]: Hmm, actually its a tad different.  The old 0.6.x `wait` code avoided errors when something like a number was passed in, if I use default arguments here it will only correct for `undefined` specifically...

Comment (Turbo87) [reply addon-test-support/ember-test-helpers/wait.js]: sure, the logic to handle non-objects would likely still need to exists, but I think putting `options = {}` in would still be good for documentation purposes

Comment (rwjblue) [reply addon-test-support/ember-test-helpers/wait.js]: updated
